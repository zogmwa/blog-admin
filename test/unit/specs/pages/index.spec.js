import Vue from 'vue'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import { mount, createLocalVue } from 'vue-test-utils'

import Index from '@/pages/index'

// 独立 Vue 构造函数
const localVue = createLocalVue()

Vue.use(ElementUI)
localVue.use(Vuex)

// 声明一个 $route 对象
const $route = {
  path: '/index',
  matched: [
    { name: '' }
  ]
}
const $router = {
  options: {
    routes: [
      {
        meta: { leaf: true },
        children: [
          { path: '/login' }
        ]
      }
    ]
  },
  push: jest.fn()
}

describe('index.vue', () => {
  let wrapper
  let state
  let store

  afterEach(() => {
    wrapper.destroy()
  })

  beforeEach(() => {
    state = {
      user: {
        gravatar: 'https://'
      }
    }
    store = new Vuex.Store({
      state
    })
  })

  it('Default data', () => {
    wrapper = mount(Index, {
      store,
      localVue,
      mocks: {
        $route,
        $router
      }
    })
    expect(wrapper.vm.defaultPath).toBe('/index')
    expect(wrapper.vm.currentPathName).toBe('')
    expect(wrapper.vm.currentPathNameParent).toBe('')
  })

  it('Render "state user"', () => {
    wrapper = mount(Index, {
      store,
      localVue
    })
    const face = wrapper.vm.$el.querySelector('.user-face-box img')
    expect(face.getAttribute('src')).toBe('https://?imageView2/1/w/36/h/36')
  })

  it('Route change', () => {
    wrapper = mount(Index, {
      store,
      localVue,
      mocks: {
        $route,
        $router
      }
    })
    wrapper.vm.$route = {
      path: '/login',
      matched: [
        { name: '' }
      ]
    }
    expect(wrapper.vm.defaultPath).toBe('/login')
    expect(wrapper.vm.currentPathName).toBe('')
    expect(wrapper.vm.currentPathNameParent).toBe('')
  })

  it('Has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
