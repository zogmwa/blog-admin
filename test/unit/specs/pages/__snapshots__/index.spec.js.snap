// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue Has the expected html structure 1`] = `
<div
  class="index"
>
  <header>
    <div
      class="logo font-futura"
    >
      <router-link
        to="/"
      >
        <img
          alt=""
          src="../assets/images/logo.png"
          width="46"
        />
         
        <span>
          三毛
        </span>
      </router-link>
    </div>
     
    <div
      class="user"
    >
      <span
        class="user-face-box"
      >
        <img
          class="user-face"
          src="https://?imageView2/1/w/36/h/36"
        />
      </span>
    </div>
  </header>
   
  <section>
    <aside
      class="nav"
    >
      <div
        class="toggle-size"
      />
       
      <ul
        class="el-menu el-menu-vertical-demo"
        role="menubar"
        style="background-color: rgb(36, 41, 46);"
        theme="dark"
      >
        <!---->
         
        <li
          class="el-menu-item"
          role="menuitem"
          style="padding-left: 20px; color: rgb(191, 201, 217); background-color: rgb(36, 41, 46);"
          tabindex="-1"
        >
          <i
            class="iconfont mar"
          />
           
          <span>
            
          </span>
        </li>
         
        <li
          class="el-menu-item"
          role="menuitem"
          style="padding-left: 20px; color: rgb(191, 201, 217); background-color: rgb(36, 41, 46);"
          tabindex="-1"
        >
          <a
            href="https://analytics.google.com"
            target="_blank"
          >
            <i
              class="iconfont mar icon-count"
            />
             
            <span>
              Google Analytics
            </span>
          </a>
        </li>
      </ul>
    </aside>
     
    <article>
      <span
        name="btn"
        tag="span"
      >
        <div
          class="breadcrumb el-col el-col-24"
        >
          <div
            aria-label="Breadcrumb"
            class="el-breadcrumb"
            role="navigation"
          >
            <span
              class="el-breadcrumb"
              name="btn"
              tag="div"
            >
              <span
                class="el-breadcrumb__item"
              >
                <span
                  class="el-breadcrumb__inner"
                  role="link"
                >
                  我的面板
                </span>
                <span
                  class="el-breadcrumb__separator"
                  role="presentation"
                >
                  /
                </span>
              </span>
               
              <span
                aria-current="page"
                class="el-breadcrumb__item"
              >
                <span
                  class="el-breadcrumb__inner"
                  role="link"
                >
                  
                </span>
                <span
                  class="el-breadcrumb__separator"
                  role="presentation"
                >
                  /
                </span>
              </span>
               
              <!---->
            </span>
          </div>
        </div>
         
        <div
          class="right-content el-col el-col-24"
        >
          <router-view
            mode="out-in"
            name="fade"
          />
        </div>
      </span>
    </article>
  </section>
</div>
`;
